serveradmin/resources/management/commands/clear_migration_log.py

```diff
@@ -0,0 +1,49 @@
+"""Serveradmin - igvm_migration_log cleanup
+
+Copyright (c) 2020 InnoGames GmbH
+"""
+from django.core.management.base import BaseCommand
+from adminapi.dataset import Query
+from adminapi.filters import Not, Empty
+from datetime import datetime
+
+class Command(BaseCommand):
+    help = 'Check for old igvm_migration_log entries and deletes them'
+
+    def handle(self, *args, **options):
+        """Entry point for command
+
+        get all hypervisors which have an entry in igvm_migration_log
+        and delete all entries which are from the previous day.
+
+        """
+
+        all_hypervisor = Query({'igvm_migration_log': Not(Empty())},
+                           ['igvm_migration_log', 'hostname'])
+
+        for hypervisor in all_hypervisor:
```

Comment:
Shouldn't each HV be `igvm_locked` for this operation?

Reply:
Why should it? The cron itself will run once in the morning and we don't interfere with anything during this cleanup.

Reply:
If cron runs during a migration, igvm won't be able to commit data with Serveradmin error "newer data available". Please don't write opportunistic software.

Reply:
This is only true for single attributes @TuxPowered42. For multi attributes this is not the case - so it would not fail.

```
>>> cron = Query({'hostname': 'somehostl'}, ['hostname', 'igvm_migration_log', 'backup_disabled']).get()
>>> igvm = Query({'hostname': 'somehost'}, ['hostname', 'igvm_migration_log', 'backup_disabled']).get()
>>> cron['igvm_migration_log'].add('1591179865 +5')
>>> igvm['igvm_migration_log'].add('1591179865 -5')
>>> cron.commit()
>>> igvm.commit()
>>> 
>>> cron['backup_disabled'] = False
>>> igvm['backup_disabled'] = False
>>> cron.commit()
>>> igvm.commit()
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/home/kofrezo/Sourcecode/serveradmin/adminapi/dataset/__init__.py", line 456, in commit
    _handle_exception(result)
  File "/home/kofrezo/Sourcecode/serveradmin/adminapi/dataset/__init__.py", line 542, in _handle_exception
    raise exception_class(result['message'])
adminapi.exceptions.DatasetError: ['Newer data available']
```

In this specific case there is also no logical conflict possible from what I understood because igvm will only add new values and the cron job will only remove values.

Reply:
In this case it's fine.